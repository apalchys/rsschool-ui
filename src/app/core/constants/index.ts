export * from './schedule';
export * from './courses';
export * from './user';
