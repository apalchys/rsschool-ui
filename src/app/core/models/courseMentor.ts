import { IUserBase } from './user';

export interface ICouseUser {
    _id: string;
    city: string;
    courseId: string;
    userId: string;
    excludeReason: string | undefined;
    isActive: boolean;
}

export interface ICourseMentor extends ICouseUser {
    menteeCapacity: number;
    mentees: Array<IUserBase>;
    preferedMentees: Array<IUserBase>;
}
