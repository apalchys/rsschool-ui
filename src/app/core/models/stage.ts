export interface IStage {
    courseId: string;
    title: string;
    startDate: number;
    endDate: number;
}

export interface IStageDocument extends IStage {
    _id: string;
}
