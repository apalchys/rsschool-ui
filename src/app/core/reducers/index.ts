import { RouterState, routerReducer } from 'react-router-redux';
import { combineReducers } from 'redux';
import { FormState, reducer as formReducer } from 'redux-form';
import { Action } from '../util';
import { CoursesState, coursesReducer } from './courses';
import { ScheduleState, scheduleReducer } from './schedule';
import { UserState, userReducer } from './user';
import { CourseState, courseReducer } from './course';

export interface RootState {
    courses: CoursesState;
    course: CourseState;
    schedule: ScheduleState;
    form: FormState;
    user: UserState;
    router: RouterState;
}

const rootReducer = combineReducers<RootState, Action<any>>({
    courses: coursesReducer,
    course: courseReducer,
    schedule: scheduleReducer as any,
    form: formReducer as any,
    user: userReducer,
    router: routerReducer as any,
});

export default rootReducer;
