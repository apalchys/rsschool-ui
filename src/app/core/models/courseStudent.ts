import { ICouseUser } from './courseMentor';
import { IUserBase, IUser } from './user';

export interface ICourseStudent extends ICouseUser {
    englishLevel: string;
    mentors: Array<IUserBase | IUser>;
    user?: IUser;
}
