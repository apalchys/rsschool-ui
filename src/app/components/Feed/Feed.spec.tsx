import { FeedActions, FeedEntities } from 'core/models';
import { shallow } from 'enzyme';
import * as React from 'react';
import Feed from './index';

describe('Feed', () => {
    it('renders correctly ', () => {
        const output = shallow(
            <Feed
                feedData={[
                    {
                        dateTime: 1530000000000,
                        actionType: FeedActions.ENROLL,
                        entityType: FeedEntities.User,
                        data: {},
                        userId: 'apalchys',
                    },
                ]}
            />,
        );
        expect(output).toMatchSnapshot();
    });

    it('renders correctly if no data', () => {
        const output = shallow(<Feed feedData={[]} />);
        expect(output).toMatchSnapshot();
    });
});
