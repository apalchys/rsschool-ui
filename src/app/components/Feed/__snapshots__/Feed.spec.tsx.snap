// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed renders correctly  1`] = `
<div>
  <h3>
    Feed
  </h3>
  <div
    className="feed-record"
    key="0"
  >
    <div
      className="feed-container"
    >
      <div>
        <div>
          2018-06-26
        </div>
        <p>
          There is some action
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Feed renders correctly if no data 1`] = `
<div>
  <h3>
    Feed
  </h3>
  <div>
    Nothing to show
  </div>
</div>
`;
